state/core_access.go

```diff
@@ -602,6 +605,41 @@ func (ca *CoreAccessor) setupTxClient(ctx context.Context, keyName string) (*use
 	)
 }
 
+func (ca *CoreAccessor) startGRPCClient(ctx context.Context) error {
+	// dial given celestia-core endpoint
+	endpoint := net.JoinHostPort(ca.coreIP, ca.port)
+	// By default, the gRPC client is configured to handle an insecure connection.
+	// If the TLS configuration is not empty, it will be applied to the client's options.
+	// If the TLS configuration is empty but the X-Token is provided,
+	// the X-Token will be applied as an interceptor along with an empty TLS configuration.
+	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+	if ca.tls != nil {
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(ca.tls)))
+	}
+
+	if interceptor := authInterceptor(ca.xtoken); interceptor != nil {
+		opts = append(opts, grpc.WithUnaryInterceptor(interceptor))
+	}
```

Comment:
nit:
Would be cleaner and would remove the check from interceptor constructor with:
if ca.xtoken != "" { add interceptor}

Reply:
done